packages/wallet/src/domain/two-player-consensus-game/index.ts

```diff
@@ -0,0 +1,199 @@
+}
+
+interface ProposalCommitment extends ConsensusGameBaseCommitment {
+  consensusCounter: 1;
+}
+
+//////////////////
+// Constructors //
+//////////////////
+
+function consensusCommitment(opts: ConsensusGameBaseCommitment): ConsensusCommitment {
+  return {
+    ...opts,
+    consensusCounter: 0,
+  };
+}
+
+function proposalCommitment(opts: ConsensusGameBaseCommitment): ProposalCommitment {
+  return {
+    ...opts,
+    consensusCounter: 1,
+  };
+}
+
+/////////////
+// Helpers //
+/////////////
+
+export function appAttributes(
+  consensusCommitmentArgs: [string, string[], string[]],
+): AppAttributes {
+  return {
+    consensusCounter: parseInt(consensusCommitmentArgs[0], 10),
+    proposedAllocation: consensusCommitmentArgs[1].map(bigNumberify).map(bn => bn.toHexString()),
+    proposedDestination: consensusCommitmentArgs[2],
+  };
+}
+
+const SolidityConsensusCommitmentType = {
+  ConsensusCommitmentStruct: {
+    consensusCounter: 'uint32',
+    proposedAllocation: 'uint256[]',
+    proposedDestination: 'address[]',
+  },
+};
+
+export function encodeAppAttributes(appAttrs: AppAttributes): Bytes {
+  return abi.encodeParameter(SolidityConsensusCommitmentType, [
+    appAttrs.consensusCounter,
+    appAttrs.proposedAllocation,
+    appAttrs.proposedDestination,
+  ]);
+}
+
+export function decodeAppAttributes(appAttrs: Bytes): AppAttributes {
+  return appAttributes(abi.decodeParameter(SolidityConsensusCommitmentType, appAttrs));
+}
+
+export function asCoreCommitment(commitment: ConsensusGameBaseCommitment): Commitment {
+  const {
+    channel,
+    commitmentType,
+    turnNum,
+    allocation,
+    destination,
+    commitmentCount,
+    consensusCounter,
+    proposedAllocation,
+    proposedDestination,
+  } = commitment;
+
+  return {
+    channel,
+    commitmentType,
+    turnNum,
+    allocation,
+    destination,
+    commitmentCount,
+    appAttributes: encodeAppAttributes({
+      consensusCounter,
+      proposedAllocation,
+      proposedDestination,
+    }),
+  };
+}
+
+export function fromCoreCommitment(commitment: Commitment): ConsensusGameBaseCommitment {
+  const { channel, turnNum, commitmentType, allocation, destination, commitmentCount } = commitment;
+  const { consensusCounter, proposedAllocation, proposedDestination } = decodeAppAttributes(
+    commitment.appAttributes,
+  );
+  return {
+    channel,
+    turnNum,
+    allocation,
+    destination,
+    commitmentCount,
+    commitmentType,
+    consensusCounter,
+    proposedAllocation,
+    proposedDestination,
+  };
+}
+
+// TODO
+// function initialize(channelDetails: ??, initialOutcome: Outcome): Proposal
+
+export function accept(commitment: ProposalCommitment): ConsensusCommitment {
+  return consensusCommitment({
+    ...commitment,
+    turnNum: commitment.turnNum + 1,
+    allocation: commitment.proposedAllocation,
+    destination: commitment.proposedDestination,
+    // TODO should proposedAllocation be conserved, or reset?
```

Comment:
Reset

Reply:
Or set to `0x0`

Reply:
The convention that we seem to have settled on is that the default values for these are *typed* but empty arrays.